### DISCUSSION TITLE
how to return a streaming response from loader?

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#6586] [Q&A] [answered]
something like this

```js
const Koa = require('koa');
const app = new Koa();

app.use(async (ctx) => {
    ctx.body = fs.createReadStream('./bigfile.tst');
});
```

**Comment by sergiodxa** [comment, marked as answer]
You can find an example here https://github.com/sergiodxa/remix-utils/blob/5b69a89f4e2c1ee8d059e69e891ba5f6cc9f3c7b/src/server/event-stream.ts#L27-L62 where I'm returning a ReadableStream.

If you're using Node you can also import `Response` from `@remix-run/node` to send Node.js streams, see https://sergiodxa.com/tutorials/send-node-read-stream-objects-from-remix-loaders, this way you could stream files read with `node:fs` module.

**Reply by gafemoyano** [reply]
Thank you!
